go.sum

```diff
@@ -181,11 +181,8 @@ github.com/btcsuite/btcd v0.0.0-20190115013929-ed77733ec07d/go.mod h1:d3C0AkH6BR
 github.com/btcsuite/btcd v0.0.0-20190315201642-aa6e0f35703c/go.mod h1:DrZx5ec/dmnfpw9KyYoQyYo7d0KEvTkk/5M/vbZjAr8=
 github.com/btcsuite/btcd v0.20.1-beta/go.mod h1:wVuoA8VJLEcwgqHBwHmzLRazpKxTv13Px/pDuV7OomQ=
 github.com/btcsuite/btcd v0.21.0-beta/go.mod h1:ZSWyehm27aAuS9bvkATT+Xte3hjHZ+MRgMY/8NJ7K94=
+github.com/btcsuite/btcd v0.22.0-beta h1:LTDpDKUM5EeOFBPM8IXpinEcmZ6FWfNZbE3lfrfdnWo=
 github.com/btcsuite/btcd v0.22.0-beta/go.mod h1:9n5ntfhhHQBIhUvlhDvD3Qg6fRUj4jkN0VB8L8svzOA=
-github.com/btcsuite/btcd v0.22.1 h1:CnwP9LM/M9xuRrGSCGeMVs9iv09uMqwsVX7EeIpgV2c=
-github.com/btcsuite/btcd v0.22.1/go.mod h1:wqgTSL29+50LRkmOVknEdmt8ZojIzhuWvgu/iptuN7Y=
-github.com/btcsuite/btcd/chaincfg/chainhash v1.0.1 h1:q0rUy8C/TYNBQS1+CGKw68tLOFYSNEs0TFnxxnS9+4U=
-github.com/btcsuite/btcd/chaincfg/chainhash v1.0.1/go.mod h1:7SFka0XMvUgj3hfZtydOrQY2mwhPclbT2snogU7SQQc=
```

Comment:
tl;dr I think this `github.com/btcsuite/btcd` `v0.23.0` is cursed and we should cancel this PR.

The `github.com/btcsuite/btcd v0.22.1` entry is being removed from `go.mod` and from the looks of `go.sum`, that causes it to regress to `v0.22.0-beta`. So this PR is supposed to be bumping it to `v0.23.0`, but it's not.

Putting `v0.23.0` in `go.mod` and then running `go mod tidy` gives an error:
```console
$ go mod tidy
go: finding module for package github.com/btcsuite/btcd/btcec
github.com/provenance-io/provenance/x/metadata/types imports
	github.com/btcsuite/btcd/btcec: module github.com/btcsuite/btcd@latest found (v0.23.0), but does not contain package github.com/btcsuite/btcd/btcec
```

Removing the line from `go.mod` and running `go mod tidy` just puts the line back in, but with version `v0.22.0-beta`.

Running `go mod why github.com/btcsuite/btcd` results in `(main module does not need package github.com/btcsuite/btcd)` even though `go mod tidy` insists on having it in the require list.

I did find that `x/metadata/types/p8e.go` imports `"github.com/btcsuite/btcd/btcec"`. So that must be where the dependency is coming from. However, as pointed out previously, according to what's in the go module system, `v0.23.0` does not seem to contain the `btcec` stuff. But it **is** contained in the source code asset for the release. So I've no idea what's up with that.

```console
$ go mod graph | grep -F ' github.com/btcsuite/btcd@' | sort -f
github.com/btcsuite/btcutil@v1.0.2 github.com/btcsuite/btcd@v0.20.1-beta
github.com/btcsuite/btcutil@v1.0.3-0.20201208143702-a53e38424cce github.com/btcsuite/btcd@v0.20.1-beta
github.com/coinbase/rosetta-sdk-go@v0.7.0 github.com/btcsuite/btcd@v0.21.0-beta
github.com/coinbase/rosetta-sdk-go@v0.7.2 github.com/btcsuite/btcd@v0.22.0-beta
github.com/cosmos/btcutil@v1.0.4 github.com/btcsuite/btcd@v0.20.1-beta
github.com/cosmos/cosmos-sdk@v0.45.1 github.com/btcsuite/btcd@v0.22.0-beta
github.com/cosmos/cosmos-sdk@v0.45.3 github.com/btcsuite/btcd@v0.22.0-beta
github.com/cosmos/ledger-cosmos-go@v0.11.1 github.com/btcsuite/btcd@v0.0.0-20190115013929-ed77733ec07d
github.com/CosmWasm/wasmd@v0.26.0 github.com/btcsuite/btcd@v0.22.0-beta
github.com/ethereum/go-ethereum@v1.10.13 github.com/btcsuite/btcd@v0.20.1-beta
github.com/ethereum/go-ethereum@v1.9.25 github.com/btcsuite/btcd@v0.0.0-20171128150713-2e60448ffcc6
github.com/tendermint/tendermint@v0.34.14 github.com/btcsuite/btcd@v0.21.0-beta
github.com/tendermint/tendermint@v0.34.19 github.com/btcsuite/btcd@v0.22.0-beta
github.com/Zilliqa/gozilliqa-sdk@v1.2.1-0.20201201074141-dd0ecada1be6 github.com/btcsuite/btcd@v0.0.0-20190315201642-aa6e0f35703c
```
So it looks like `v0.22.0-beta` is the most recent one needed indirectly.

Also:
```console
> go get github.com/btcsuite/btcd/btcec
go: module github.com/btcsuite/btcd@upgrade found (v0.23.0), but does not contain package github.com/btcsuite/btcd/btcec
```

So I'm guessing something got messed up on their release.

In any case, this PR is not doing the bump it thinks it's doing.